### PR TITLE
Specify valid flags in JSON

### PR DESCRIPTION
**Pull request by mugling**
See also #18338

Valid flags are specified via `json_flag` along with the contexts in which they are valid. At present `contexts` and `conflicts` are unchecked but a further PR will extend the linter to reject invalid flags.

```
[
  {
    "id": "BELTED",
    "type": "json_flag",
    "context": [ "ARMOR", "armor_data" ],
    "description": "Layer for backpacks and things worn over outerwear.",
    "info": "This gear is <info>strapped</info> onto you.",
    "conflicts": [ "OUTER", "SKINTIGHT", "WAIST" ]
  },
  {
    "id": "kevlar_padded",
    "type": "json_flag",
    "context": [ "ARMOR", "armor_data" ],
    "info": "This gear has Kevlar inserted into strategic locations to <good>increase protection</good> with some <bad>increase to encumbrance</bad>.",
    "pseudo": true
  }
]
```

Automatic generation of documentation is supported via `flagdoc.pl`. Substitution contexts are specified in `sections.conf` and matched against `@section@` in `template.in`.

### CODE REVIEW

**Comment by mugling** [comment]
Ready. Next PR will extend coverage to include all known flag fields. Third PR will enable linting.

File: data/json/flags.json [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    "id": "HYGROMETER",
+    "type": "json_flag",
+    "context": [ "ARMOR", "armor_data" ],
+    "description": "This gear is equipped with an accurate hygrometer (which is used to measure humidity)."
+  },
+  {
+    "id": "NO_QUICKDRAW",
+    "type": "json_flag",
+    "context": [ "ARMOR", "armor_data" ],
+    "description": "Don't offer to draw items from this holster when the fire key is pressed whilst the players hands are empty"
```
Comment (illi-kun) [inline data/json/flags.json]: Please add a period at the end of the sentence.

File: data/json/flags.json [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    "context": [ "ARMOR", "armor_data" ],
+    "info": "This clothing will <good>protect</good> you from harm and withstand <info>a lot of abuse</info>.",
+    "conflicts": [ "FRAGILE" ]
+  },
+  {
+    "id": "SUN_GLASSES",
+    "type": "json_flag",
+    "context": [ "ARMOR", "armor_data" ],
+    "description": "Prevents glaring when in sunlight.",
+    "info": "This clothing keeps the <info>glare</info> out of your eyes"
```
Comment (illi-kun) [inline data/json/flags.json]: period at the end, please.

File: data/json/flags.json [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    "type": "json_flag",
+    "context": [ "ARMOR", "armor_data" ],
+    "description": "Acts as a watch and allows the player to see actual time."
+  },
+  {
+    "id": "WATERPROOF",
+    "type": "json_flag",
+    "context": [ "ARMOR", "armor_data" ],
+    "description": "Prevents the covered body-part(s) from getting wet in any circumstance.",
+    "info": "This clothing <info>won't let water through</info>.  Unless you jump in the river or something like that"
```
Comment (illi-kun) [inline data/json/flags.json]: and here too.

**Comment by mugling** [comment]
@illi-kun anything further?

File: data/json/flags.json (hunk 3 above)
Comment (mugling) [inline data/json/flags.json]: done

File: data/json/flags.json (hunk 2 above)
Comment (mugling) [inline data/json/flags.json]: done

File: data/json/flags.json (hunk 1 above)
Comment (mugling) [inline data/json/flags.json]: done

**Comment by illi-kun** [comment]
@mugling 
> anything further

not from my side but I want to have another pair of eyes on this.

> github has code review

I'll try this new feature next time, thanks for note.

**Comment by mugling** [comment]
> I'll try this new feature next time, thanks for note.

I'm not yet sure how it works though

**Pull request merged**